tools/configuration-converter.sh

```diff
@@ -83,41 +83,37 @@ while [[ $# -gt 0 ]]; do
     esac
 done
 
-
 if ! command -v yq &>/dev/null; then
     echo "Error: 'yq' is not installed. Please install it before running the script."
     exit 1
 fi
 
 function add_mandatory_fields(){
-    # Base yq command without the version field
-    yq_cmd=".apiVersion = \"sailoperator.io/v1\" 
-            | .kind = \"Istio\" 
-            | (select(.spec.meshConfig) | .spec.values.meshConfig) = .spec.meshConfig 
-            | (select(.spec.values.istio_cni) | .spec.values.pilot.cni) = .spec.values.istio_cni 
-            | .metadata.name = \"default\" 
-            | .spec.namespace = \"$NAMESPACE\" 
-            | del(.spec.values.istio_cni)
-            | del(.spec.meshConfig)
-            | del(.spec.hub) 
-            | del(.spec.tag) 
-            | del(.spec.values.gateways)"
-    
     # If VERSION is not empty, add .spec.version
     if [[ -n "$VERSION" ]]; then
-        yq_cmd="$yq_cmd | .spec.version = \"$VERSION\""
+        yq -i ".spec.version = \"$VERSION\"" "$OUTPUT"  
     fi
```

Comment:
nit: I'd prefer to see this *after* the main yq command, as this here is just an additional step; the main part of this function is the large yq command, so it should come first.